src/error.ts

```diff
@@ -1246,11 +1253,16 @@ export function needsRetryableWriteLabel(error: Error, maxWireVersion: number):
   }
 
   if (error instanceof MongoWriteConcernError) {
-    return RETRYABLE_WRITE_ERROR_CODES.has(error.result.writeConcernError.code ?? error?.code ?? 0);
+    if (!(serverType === 'Mongos' && maxWireVersion < 9)) {
+      return RETRYABLE_WRITE_ERROR_CODES.has(
+        error.result.writeConcernError.code ?? Number(error.code) ?? 0
+      );
+    }
+    return RETRYABLE_WRITE_ERROR_CODES.has(Number(error.originalCode) ?? 0);
   }
 
-  if (error instanceof MongoError && typeof error.code === 'number') {
-    return RETRYABLE_WRITE_ERROR_CODES.has(error.code);
+  if (error instanceof MongoError) {
+    return RETRYABLE_WRITE_ERROR_CODES.has(Number(error.code));
```

Comment:
Is the coercion to number here just because it seems correct or do tests fail without it? I agree this seems correct but just curious.

Reply:
The coercion to number is because error.code can be a string or a number as define on `MongoError`.